### PR TITLE
Upgrade to go version 1.21

### PR DESCRIPTION
**Pull request by ashmeenkaur**
### Description
Upgrade to go version 1.21 as it fixes security vulnerabilities in the older version.

### Link to the issue in case of a bug fix.
https://github.com/GoogleCloudPlatform/gcsfuse/issues/1288

### Testing details
1. Manual - basic sanity checks with go run command
2. Unit tests - NA
3. Integration tests 
   - ran integration tests on KOKORO.
   - ran integration tests on deb and rpm package via louhi on all OS types.

### CODE REVIEW

File: tools/containerize_gcsfuse_docker/Dockerfile [hunk 1]
```diff
@@ -49,8 +49,8 @@ RUN go get -d ${GCSFUSE_REPO}
 
 WORKDIR ${GCSFUSE_PATH}
 # Branch name for building through a particular branch.
-ARG BRANCH_NAME="v${GCSFUSE_VERSION}"
```
Comment (vadlakondaswetha) [inline tools/containerize_gcsfuse_docker/Dockerfile:52]: why is this code changed?

**Review by vadlakondaswetha — COMMENTED** [review]
can you perf and integration tests on this PR and see if the go version works without any issue.

File: tools/containerize_gcsfuse_docker/Dockerfile (hunk 1 above)
Comment (ashmeenkaur) [reply tools/containerize_gcsfuse_docker/Dockerfile]: Changed it to run integration tests on louhi. Reverted the change.

**Comment by ashmeenkaur** [comment]
> can you perf and integration tests on this PR and see if the go version works without any issue.

The integration tests ran successfully. Running perf tests with target as sept-release branch needs changes in the perf test script.

**Comment by ashmeenkaur** [comment]
> > can you perf and integration tests on this PR and see if the go version works without any issue.
> 
> The integration tests ran successfully. Running perf tests with target as sept-release branch needs changes in the perf test script.

No Significant difference between master and PR perf results. [link](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/43f311dd-61f4-49ea-9c48-9e56ae0b6fc7/log)

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**